src/main/java/org/openrewrite/java/spring/boot3/MigrateWebMvcTagsToObservationConvention.java

```diff
@@ -0,0 +1,285 @@
+                              "    @Override\n" +
+                              "    public KeyValues getLowCardinalityKeyValues(ServerRequestObservationContext context) {\n" +
+                              "        KeyValues values = super.getLowCardinalityKeyValues(context);\n" +
+                              "        return values;" +
+                              "    }\n" +
+                              "}";
+                J.ClassDeclaration newClassDeclaration = JavaTemplate.builder(tmpl)
+                        .contextSensitive()
+                        .javaParser(JavaParser.fromJavaVersion().classpathFromResources(ctx, "micrometer-commons", "spring-web-6.+", "jakarta.servlet-api"))
+                        .imports(DEFAULTSERVERREQUESTOBSERVATIONCONVENTION_FQ, KEYVALUES_FQ, HTTPSERVLETREQUEST_FQ, HTTPSERVLETRESPONSE_FQ, SERVERREQUESTOBSERVATIONCONVENTION_FQ)
+                        .build()
+                        .apply(getCursor(), classDecl.getCoordinates().replace());
+
+                J.ClassDeclaration finalNewClassDeclaration = newClassDeclaration;
+                newClassDeclaration = newClassDeclaration
+                        .withLeadingAnnotations(classDecl.getLeadingAnnotations())
+                        .withModifiers(classDecl.getModifiers())
+                        .withPrefix(classDecl.getPrefix())
+                        .withBody(newClassDeclaration.getBody().withStatements(ListUtils.map(classDecl.getBody().getStatements(), stmt -> {
+                            if (stmt instanceof J.MethodDeclaration && ((J.MethodDeclaration) stmt).getSimpleName().equals("getTags")) {
+                                J.MethodDeclaration md = (J.MethodDeclaration) finalNewClassDeclaration.getBody().getStatements().get(0);
+                                md = md.withPrefix(stmt.getPrefix());
+                                //noinspection DataFlowIssue
+                                return md.withBody(md.getBody().withStatements(ListUtils.insertAll(md.getBody().getStatements(), md.getBody().getStatements().size() - 1, getTagsBodyStatements)));
+                            }
+                            return stmt;
+                        })));
+                maybeAddImport(DEFAULTSERVERREQUESTOBSERVATIONCONVENTION_FQ);
+                maybeAddImport(KEYVALUE_FQ);
+                maybeAddImport(KEYVALUES_FQ);
+                maybeAddImport(SERVERREQUESTOBSERVATIONCONVENTION_FQ);
+                maybeRemoveImport(HTTPSERVLETREQUEST_FQ);
+                maybeRemoveImport(HTTPSERVLETRESPONSE_FQ);
+                maybeRemoveImport(TAG_FQ);
+                maybeRemoveImport(TAGS_FQ);
+                maybeRemoveImport(WEBMVCTAGS_FQ);
+                maybeRemoveImport(WEBMVCTAGSPROVIDER_FQ);
+                return (J.ClassDeclaration) super.visitClassDeclaration(newClassDeclaration, ctx);
```

Comment:
With the super call being down here, I think we should update the cursor with `newClassDeclaration`, just so that there isn't any surprises if anyone in another visit method calls `getParent()` and expects to find the updated class declaration.

Alternatively we could maybe do the super call first?